NPE with dart entity

---- Minecraft Crash Report ----

WARNING: coremods are present:
  llibrary (llibrary-core-1.0.8-1.12.2.jar)
Contact their authors BEFORE contacting forge

// You're mean.

Time: 9/2/18 4:54 PM
Description: Exception ticking world

io.netty.handler.codec.EncoderException: java.lang.NullPointerException
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:106)
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:704)
	at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1017)
	at io.netty.channel.AbstractChannel.write(AbstractChannel.java:286)
	at io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:341)
	at net.minecraftforge.fml.common.network.FMLEmbeddedChannel.generatePacketFrom(FMLEmbeddedChannel.java:72)
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.getEntitySpawningPacket(FMLNetworkHandler.java:134)
	at net.minecraft.entity.EntityTrackerEntry.func_151260_c(EntityTrackerEntry.java:501)
	at net.minecraft.entity.EntityTrackerEntry.func_73117_b(EntityTrackerEntry.java:380)
	at net.minecraft.entity.EntityTracker.func_85172_a(EntityTracker.java:392)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187272_b(PlayerChunkMapEntry.java:162)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:165)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at net.minecraft.network.PacketBuffer.func_150788_a(PacketBuffer.java:326)
	at net.minecraftforge.fml.common.network.ByteBufUtils.writeItemStack(ByteBufUtils.java:180)
	at org.jurassicraft.server.entity.TranquilizerDartEntity.writeSpawnData(TranquilizerDartEntity.java:72)
	at net.minecraftforge.fml.common.network.internal.FMLMessage$EntitySpawnMessage.toBytes(FMLMessage.java:213)
	at net.minecraftforge.fml.common.network.internal.FMLRuntimeCodec.encodeInto(FMLRuntimeCodec.java:42)
	at net.minecraftforge.fml.common.network.internal.FMLRuntimeCodec.encodeInto(FMLRuntimeCodec.java:32)
	at net.minecraftforge.fml.common.network.FMLIndexedMessageToMessageCodec.encode(FMLIndexedMessageToMessageCodec.java:77)
	at io.netty.handler.codec.MessageToMessageCodec$1.encode(MessageToMessageCodec.java:67)
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:88)
	... 22 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:106)
	at io.netty.handler.codec.MessageToMessageCodec.write(MessageToMessageCodec.java:116)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:704)
	at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1017)
	at io.netty.channel.AbstractChannel.write(AbstractChannel.java:286)
	at io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:341)
	at net.minecraftforge.fml.common.network.FMLEmbeddedChannel.generatePacketFrom(FMLEmbeddedChannel.java:72)
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.getEntitySpawningPacket(FMLNetworkHandler.java:134)
	at net.minecraft.entity.EntityTrackerEntry.func_151260_c(EntityTrackerEntry.java:501)
	at net.minecraft.entity.EntityTrackerEntry.func_73117_b(EntityTrackerEntry.java:380)
	at net.minecraft.entity.EntityTracker.func_85172_a(EntityTracker.java:392)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187272_b(PlayerChunkMapEntry.java:162)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:165)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)

-- Affected level --
Details:
	Level name: New World
	All players: 1 total; [EntityPlayerMP['TOG_SinfulDark'/408, l='New World', x=-277.80, y=77.11, z=240.77]]
	Chunk stats: ServerChunkCache: 810 Drop: 0
	Level seed: -4792871786917087568
	Level generator: ID 00 - default, ver 1. Features enabled: true
	Level generator options: 
	Level spawn location: World: (64,64,133), Chunk: (at 0,4,5 in 4,8; contains blocks 64,0,128 to 79,255,143), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 1927038 game time, 1369 day time
	Level dimension: 0
	Level storage version: 0x04ABD - Anvil
	Level weather: Rain time: 11053 (now: false), thunder time: 169426 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 3368330656 bytes (3212 MB) / 4026531840 bytes (3840 MB) up to 10737418240 bytes (10240 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx10G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=16M
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
	FML: MCP 9.42 Powered by Forge 14.23.4.2705 21 mods loaded, 21 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID                   | Version                        | Source                                        | Signature                                |
	|:--------- |:-------------------- |:------------------------------ |:--------------------------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft            | 1.12.2                         | minecraft.jar                                 | None                                     |
	| UCHIJAAAA | mcp                  | 9.42                           | minecraft.jar                                 | None                                     |
	| UCHIJAAAA | FML                  | 8.0.99.99                      | forge-1.12.2-14.23.4.2705.jar                 | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge                | 14.23.4.2705                   | forge-1.12.2-14.23.4.2705.jar                 | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | betterbuilderswands  | 0.11.1                         | Better-Builders-Wands-Mod-1.12.jar            | None                                     |
	| UCHIJAAAA | chancecubes          | 1.12.1-3.0.1.208               | Chance-Cubes-Mod-1.12.1.jar                   | None                                     |
	| UCHIJAAAA | cookingforblockheads | 6.4.44                         | Cooking-for-Blockheads-Mod-1.12.2.jar         | None                                     |
	| UCHIJAAAA | enhancedvisuals      | 1.0                            | EnhancedVisuals v1.1.4 mc1.12.2.jar           | None                                     |
	| UCHIJAAAA | grapplemod           | 1.12.2-v10                     | Grappling-Hook-Mod-1.12.2.jar                 | None                                     |
	| UCHIJAAAA | jei                  | 4.11.0.206                     | jei_1.12.2-4.11.0.206.jar                     | None                                     |
	| UCHIJAAAA | jeresources          | 0.8.5.26                       | Just-Enough-Resources-Mod-1.12.2.jar          | None                                     |
	| UCHIJAAAA | lucky                | 7.5.0                          | Lucky-Block-Mod-1.12.zip                      | None                                     |
	| UCHIJAAAA | machinemod           | 1.12.2-1.1-Pre-ALPHA-Build-256 | MachineMod-1.12.2-1.1-Pre-ALPHA-Build-256.jar | None                                     |
	| UCHIJAAAA | cfm                  | 5.3.1                          | MrCrayfishs-Furniture-Mod-1.12.2.jar          | None                                     |
	| UCHIJAAAA | ropebridge           | 2.0.5                          | ropebridge-1.12-2.0.7.jar                     | None                                     |
	| UCHIJAAAA | bspkrscore           | at_0.9                         | TreeCapitator-Mod-1.12.jar                    | None                                     |
	| UCHIJAAAA | treecapitator        | at_0.9                         | TreeCapitator-Mod-1.12.jar                    | None                                     |
	| UCHIJAAAA | veinminer            | 0.38.1                         | Vein-Miner-Mod-1.12.jar                       | None                                     |
	| UCHIJAAAA | veinminermodsupport  | 0.38.1                         | Vein-Miner-Mod-1.12.jar                       | None                                     |
	| UCHIJAAAA | llibrary             | 1.7.15                         | llibrary-1.7.15-1.12.2.jar                    | b9f30a813bee3b9dd5652c460310cfcd54f6b7ec |
	| UCHIJAAAA | jurassicraft         | 2.1.17                         | JurassiCraft-1.12.2-2.1.17 (1).jar            | None                                     |

	Loaded coremods (and transformers): 
llibrary (llibrary-core-1.0.8-1.12.2.jar)
  net.ilexiconn.llibrary.server.core.plugin.LLibraryTransformer
  net.ilexiconn.llibrary.server.core.patcher.LLibraryRuntimePatcher
	GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	Profiler Position: N/A (disabled)
	Player Count: 1 / 8; [EntityPlayerMP['TOG_SinfulDark'/408, l='New World', x=-277.80, y=77.11, z=240.77]]
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fml,forge'

--- a/CHANGELOG.txt
+++ b/CHANGELOG.txt
@@ -1,3 +1,4 @@
-*Fixed the dinosaur's order gui
-*The helicopter is now capable of flying on multiplayer servers
-*Fixed color rendering on tour rails
+*Fixed that spawning action figures of a certain gender didn't work on servers
+*Fixed that dinosaurs weren't rendered correctly in mob spawners
+*Fixed that decayable JC leaves are loosing that property after some time
+*Fixed a rare crash problem with the dart gun

--- a/src/main/java/org/jurassicraft/server/entity/TranquilizerDartEntity.java
+++ b/src/main/java/org/jurassicraft/server/entity/TranquilizerDartEntity.java
@@ -69,12 +69,14 @@ public class TranquilizerDartEntity extends EntityThrowable implements IEntityAd
 
     @Override
     public void writeSpawnData(ByteBuf buffer) {
-	ByteBufUtils.writeItemStack(buffer, stack);
+	if(stack != null)
+		ByteBufUtils.writeItemStack(buffer, stack);
     }
 
     @Override
     public void readSpawnData(ByteBuf additionalData) {
-	stack = ByteBufUtils.readItemStack(additionalData);
+	if(stack != null)
+		stack = ByteBufUtils.readItemStack(additionalData);
     }
     
 }

